src/supervisor.sbvr

```diff
@@ -0,0 +1,461 @@
+		Necessity: each image is a build of exactly one service.
+
+	Fact type: service has status
+		Necessity: each service has exactly one status.
+		Definition: "Running" or "Stopping" or "Stopped"
+		-- TODO: these are partially taken from v2 API reference which may not be ideal. Decide on status strings for v3.
+
+	Fact type: application has service
+		Necessity: each application has at least one service.
+		Necessity: each service is managed by at most one application.
+
+
+
+-- RELEASE
+
+Term: release
+
+	Fact type: image is part of release
+		Synonymous Form: release contains image
+
+	Fact type: device is running release
+		Synonymous Form: release is running on device
+		Necessity: each device is running at least one release.
+		-- Changed from "at most one release" to "at least" for multi-app -- valid?
+
+	Fact type: release belongs to application
+		Synonymous Form: application owns release
+		Necessity: each release belongs to exactly one application.
+
+
+
+-- APPLICATION
+
+Term: api port
+	Concept Type: Integer (Type)
+
+Term: api secret
+	Concept Type: Short Text (Type)
+
+Term: commit
+	Concept Type: Short Text (Type)
+
+Term: application
+
+	Fact type: application has slug
+	-- Not currently provided to Supervisor, optional, but would be nice for LocalUI
+		Necessity: each application has exactly one slug
+		Necessity: each slug is of exactly one application
+
+	Fact type: application has api port
+		Necessity: each application that has a slug that is equal to "balena-os/balena-supervisor", has exactly one api port that is equal to 48484.
+
+	Fact type: application has name
+	-- Not currently provided to Supervisor, optional, but would be nice for LocalUI
+		Necessity: each application has exactly one name.
+		Necessity: each application has an app name that has a Length (Type) that is greater than or equal to 4 and is less than or equal to 100.
+
+	Fact type: application has uuid
+	-- Optional: wait for v3 target state?
+		Necessity: each application has exactly one uuid.
+		Necessity: each uuid is of exactly one application.
+		Necessity: each application has a uuid that has a Length (Type) that is equal to 32.
+
+	Fact type: application is host
+
+	Fact type: application is healthy
+	-- GET /v1/healthy
+		Necessity: each application that has a slug that is equal to "balena-os/balena-supervisor" is healthy or not healthy.
+		-- Not sure about syntax / validity for "healthy" boolean
+	
+	Fact type: application has download progress
+	-- Maybe should be computed term of all image download progresses?
+		Necessity: each application has at most one download progress.
+
+	Fact type: application should be running release
+		Synonymous Form: release should be running on application
+		Necessity: each application should be running at most one release.
+
+	Fact type: application has version
+		Necessity: each application that has a slug that is equal to "balena-os/balena-supervisor" has exactly one version.
```

Comment:
Isn't version a property of the release? Why only the supervisor needs to have an app version ?

Reply:
Fixed, removed this Fact and included the following:
```
Fact type: release has version
		Necessity: each release has at most one version.
```